docs/GDI-SOP_github-introduction-for-maintainers.md

```diff
@@ -0,0 +1,193 @@
+2. [Understanding the Repository](#understanding-the-repository)
+3. [Issues](#issues)
+4. [Branches](#branches)
+5. [Pull Requests (PRs)](#pull-requests)
+6. [Workflows and Linting](#workflows-and-linting)
+7. [General Recommendations](#general-recommendations)
+8. [Further Reading](#further-reading)
+
+## What is GitHub?
+
+GitHub is a cloud-based platform that allows for **collaborative development of projects through version control**. For this repository, GitHub allows multiple maintainers and contributors to work on the **Standard Operating Procedures (SOPs)** simultaneously, ensuring that changes are tracked and reviewed before they are added to the `main` branch. Each repository is made up of a combination of code files, documentation, and workflows that automate certain actions (e.g., verifying formatting of SOPs).
+
+If you are familiar with Google Drive, OneDrive, or Dropbox, you can think of GitHub as a similar platform. Even though the differences are notable, for the purpose of this introduction, you can imagine that when you modify a GitHub repository, you are modifying a directory (i.e., a folder) shared with contributors through the internet. We will promptly discuss the features of GitHub that go beyond just storing files.
+
+### Key GitHub Concepts:
+- **Repository** (_blue_): A directory containing project files. In this case, the GDI SOP repository stores the SOPs and associated metadata.
+- **Branch** (_yellow_): A separate version of the repository where you can work on changes without affecting the main codebase (also known as the `main` branch).
+- **Commit** (_orange_): A record of changes made to files. As the name implies, when you "commit" any change to a file, you are committing those changes to the branch where you are working.
+- **Pull Request (PR)** (_pink_): A request to merge changes from one branch to another.
+- **Issue** (_green_): A way to track tasks, improvements, or bugs in the repository.
+
+![Basic layout of GitHub](images/GDI-SOP_github-introduction-for-maintainers_basic-layout.png)
+
+We will elaborate on these key concepts further in the sections below.
+
+### Example of GitHub usage
+See the diagram below for a simple example of how changes are propagated through the branches of this repository.
+
+From left to right:
+- The GDI SOP repository (`main` branch) was created.
+- New changes (e.g., documentation files) were added.
+- A new branch (`dev`) was created to hold 'new features' until they were ready to be part of the `main` branch.
+- A new SOP was scheduled, so another branch, `new-SOP`, was created.
+- After adding and reviewing the new SOP, it was merged into the `dev` branch.
+- More work was done on `dev` (e.g., adding new documents, SOPs, modifying content). Keep in mind that so far, the `main` branch remained unaltered.
+- Eventually, when the new changes in `dev` (including the "new SOP") were ready, `dev` was merged into `main`.
+
+````mermaid
+gitGraph
```

Comment:
I was hoping to represent how branches, commits and PRs work through this diagram. The "gitGraph" is just a type of mermaid diagram. The rendered diagram exemplifies how branches are used to contain features and then bring them back to ``main``. I think without the graph... it's even more confusing, no?

Reply:
To be honest ... I think this one was much easier to grasp it ... I think that you refer to the same thing! 
![image](https://github.com/user-attachments/assets/cf532dfb-201e-4ec7-b08c-730bd857c590)

Reply:
Ah I see where the issue may be: I think that what you were commenting was on the text itself, without it being rendered as a diagram. The ``gitGraph`` piece is the "text" definition of the diagram, but the diagram is rendered in markdown.

In other words, if you see this raw file, you'll find the mermaid code block, but that's not what the file looks like when rendered. Similar to how other formattings are encoded in raw markdown but rendered later.

Check [**this file section**](https://github.com/GenomicDataInfrastructure/standard-operating-procedures/blob/933e047878cf1859c298044243036df9e316b4dd/docs/GDI-SOP_github-introduction-for-maintainers.md#example-of-github-usage) in the rendered document, in case that's why it was so confusing to read. I agree that the text itself is incredibly confusing, but the text renders into the diagram 🙏

Reply:
Ok! This was exactly the problem. I could not see the diagram, only the mermaid code block and I did not understand so much from the block !! When I checked "this file section"  I came to the diagram which is much more easier to understand and grasp it! There is no problem ... it is solved! :-)